projects/packages/videopress/src/client/admin/hooks/use-query-string-pages/index.ts

```diff
@@ -0,0 +1,38 @@
+/**
+ * External dependencies
+ */
+import { useHistory } from 'react-router-dom';
+
+/**
+ * Uses the history to manipulate the URL pagination parameters.
+ *
+ * @returns {object} - Object containing useful handlers for URL pagination
+ */
+const useQueryStringPages = () => {
+	const history = useHistory();
+	const setPageOnURL = page => {
+		const searchFragment = page > 1 ? `?page=${ page }` : '';
+		if ( searchFragment !== history.location.search ) {
```

Comment:
When adding a second fragment, like a filter, we would need to check if the history search _contains_ the search fragment (without `?`).